pgmpy/factors/Factor.py

```diff
@@ -211,18 +211,18 @@ def _marginalize_single_variable(self, variable):
             name of variable to be marginalized
 
         """
-        index = list(self.variables.keys()).index(variable)
-        cum_cardinality = (np.product(self.cardinality) /
-                           np.concatenate(([1], np.cumprod(self.cardinality)))).astype(np.int64, copy=False)
-        num_elements = cum_cardinality[0]
-        sum_index = [j for i in range(0, num_elements,
-                                      cum_cardinality[index])
-                     for j in range(i, i+cum_cardinality[index+1])]
-        marg_factor = np.zeros(num_elements/self.cardinality[index])
-        for i in range(self.cardinality[index]):
-            marg_factor += self.values[np.array(sum_index) +
-                                       i*cum_cardinality[index+1]]
-        return marg_factor
+        marginalize_index = np.where(np.in1d(self.scope(),variable))
+        assign = np.array(self.cardinality)
+        assign[marginalize_index] = -1
+        marg_factor = []
+        for i in itertools.product(*[range(index) for index in self.cardinality[np.where(assign!=-1)[0]]]):
+            temp_sum = 0
+            sum_variable = np.array(i)
+            sum_variable = np.insert(sum_variable,np.int(marginalize_index[0]),-1)
```

Comment:
There should be a space after **,**. It should be 
```python
np.insert(sum_variable, np.int(marginalize_index[0]), -1)
```
Use ```pep8``` checker to check for all PEP8 errors. Please make sure that your code doesn't have any PEP8 violations (except for max line length which we had to 120 characters) before committing it. Use ```pep8``` checker as
```
pep8 file_name.py --max-line-length=120
```